src/generators/Sequence.js

```diff
@@ -4,27 +4,30 @@
 
 import Generator from './Generator';
 
-class Sequence extends Generator {
+export default class Sequence extends Generator {
   static sequences = {};
-  id = '';
 
-  constructor(factoryGirl, id, callback = null) {
+  constructor(factoryGirl, id = null, callback = null) {
     super(factoryGirl);
-
-    if (typeof id !== 'string') {
-      throw new Error('Invalid sequence key passed');
+    if (typeof id === 'function') {
+      callback = id;
+      id = null;
     }
-
-    this.id = id;
-
-    Sequence.sequences[id] = Sequence.sequences[id] || 1;
+    this.id = id || generateId();
     this.callback = callback;
+    Sequence.sequences[this.id] = Sequence.sequences[this.id] || 1;
   }
-
   generate() {
-    const count = Sequence.sequences[this.id]++;
-    return Promise.resolve(this.callback ? this.callback(count) : count);
+    const next = Sequence.sequences[this.id]++;
+    return this.callback ? this.callback(next) : next;
   }
 }
 
-export default Sequence;
+function generateId() {
+  let id;
+  let i = 0;
+  do {
+    id = `_${i++}`;
+  } while (id in Sequence.sequences);
+  return id;
+}
```

Comment:
I suppose we should mention in the docs that sequence-id is optional only when using object initialiser. For function initialiser the sequence-id should be mandatory, otherwise it will keep starting from `1` for each attrs/build/create call.